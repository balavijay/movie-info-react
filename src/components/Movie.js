import React from 'react';

// Components
import Navigation from './elements/Navigation/Navigation';
import MovieInfo from './elements/MovieInfo/MovieInfo';
import Actor from './elements/Actor/Actor';
import Grid from './elements/Grid/Grid';
import Spinner from './elements/Spinner/Spinner';

import { useMovieFetch } from './hooks/useMovieFetch';

const Movie = ({ movieId }) => {
  const [movie, loading, error] = useMovieFetch(movieId);
  console.log(movie);
  if (error) return <div>Something went wrong ...</div>;
  if (loading) return <Spinner />;

  return (
  <>
    <Navigation movie={movie.original_title} />
    <MovieInfo movie={movie} />

    <Grid header="Cast">
      {movie.actors && movie.actors.map(actor => (
        <Actor key={actor.credit_id} actor={actor} />
      ))}     
    </Grid>
  </>
  )
};

export default Movie;