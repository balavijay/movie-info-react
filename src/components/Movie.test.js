import React from 'react';
import { shallow, mount } from 'enzyme';

import Movie from './Movie';

describe ("Movie Component " , () => {
   const  app = shallow(<Movie />);
    test("Component rendered" , () => {
        expect(app).toMatchSnapshot();
    });

    
});