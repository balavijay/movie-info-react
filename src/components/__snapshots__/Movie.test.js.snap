// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Component  Component rendered 1`] = `ShallowWrapper {}`;
